pkg/resource/function/hooks.go

```diff
@@ -614,63 +621,114 @@ func (rm *resourceManager) setResourceAdditionalFields(
-					ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination = getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess.Destination
-				}
-			}
-		} else {
-			ko.Spec.FunctionEventInvokeConfig.DestinationConfig = nil
-		}
-		if getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds != nil {
-			ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds = getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds
-		} else {
-			ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds = nil
-		}
-		if getFunctionEventInvokeConfigOutput.DestinationConfig != nil {
-			ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts = getFunctionEventInvokeConfigOutput.MaximumRetryAttempts
-		} else {
-			ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts = nil
-		}
+		// creating FunctionEventInvokeConfig object to store the values returned from `Get` call
+		cloudFunctionEventInvokeConfig := &svcapitypes.PutFunctionEventInvokeConfigInput{}
+		cloudFunctionEventInvokeConfig.DestinationConfig = &svcapitypes.DestinationConfig{}
+		cloudFunctionEventInvokeConfig.DestinationConfig.OnFailure = &svcapitypes.OnFailure{}
+		cloudFunctionEventInvokeConfig.DestinationConfig.OnSuccess = &svcapitypes.OnSuccess{}
+		cloudFunctionEventInvokeConfig.DestinationConfig.OnFailure.Destination = getFunctionEventInvokeConfigOutput.DestinationConfig.OnFailure.Destination
+		cloudFunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination = getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess.Destination
+		cloudFunctionEventInvokeConfig.MaximumEventAgeInSeconds = getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds
+		cloudFunctionEventInvokeConfig.MaximumRetryAttempts = getFunctionEventInvokeConfigOutput.MaximumRetryAttempts
+		ko.Spec.FunctionEventInvokeConfig = cloudFunctionEventInvokeConfig
+	}
+	return nil
+}
+
+// getFunctionEventInvokeConfig will describe the fields that are
+// custom to the Function resource
+func (rm *resourceManager) setFunctionEventInvokeConfig(
+	ctx context.Context,
+	ko *svcapitypes.Function,
+) (err error) {
+	var getFunctionEventInvokeConfigOutput *svcsdk.GetFunctionEventInvokeConfigOutput
+	getFunctionEventInvokeConfigOutput, err = rm.sdkapi.GetFunctionEventInvokeConfigWithContext(
+		ctx,
+		&svcsdk.GetFunctionEventInvokeConfigInput{
+			FunctionName: ko.Spec.Name,
+		},
+	)
+	rm.metrics.RecordAPICall("GET", "GetFunctionEventInvokeConfig", err)
+
+	err = rm.setFunctionEventInvokeConfigFromResponse(ko, getFunctionEventInvokeConfigOutput, err)
+	if err != nil {
+		return err
 	}
+
+	return nil
+}
+
+// setResourceAdditionalFields will describe the fields that are not return by
+// API calls
+func (rm *resourceManager) setResourceAdditionalFields(
+	ctx context.Context,
+	ko *svcapitypes.Function,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.setResourceAdditionalFields")
+	defer exit(err)
```

Comment:
I wonder if we should have a section like this with `rlog.Trace`/`defer` for all the above new functions `setSomething()`. 

It sounds like it would help debugging

Reply:
Done